mlir/test/Target/rocdl.mlir

```diff
@@ -1,4 +1,6 @@
 // RUN: mlir-translate -mlir-to-rocdlir %s | FileCheck %s
+// CHECK: target datalayout = "e-p:64:64-p1:64:64-p2:32:32-p3:32:32-p4:64:64-p5:32:32-p6:32:32-i64:64-v16:16-v24:32-v32:32-v48:64-v96:128-v192:256-v256:256-v512:512-v1024:1024-v2048:2048-n32:64-S32-A5-ni:7"
```

Comment:
we probably don't need these checks around datalayout and triple.

I'll be submitting another PR where we deliberately change width of pointers in generic address space to 32-bits so we tame the width of MLIR `index` type be the same as `i32` so avoid unnecessary integer upcast/downcast while doing GEP/load/store.